test/E2ETest/Microsoft.Test.E2E.AspNet.OData/BulkOperation/BulkInsertDataModel.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Builder;
+using Purchasing;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Data.Entity;
+
+namespace Microsoft.Test.E2E.AspNet.OData.BulkInsert1
+{
+    public class Employee
+    {
+        [Key]
+        public int ID { get; set; }
+        public String Name { get; set; }        
+        public List<Skill> SkillSet { get; set; }
+        public Gender Gender { get; set; }
+        public AccessLevel AccessLevel { get; set; }
+        
+        [AutoExpand]
+        public List<Friend> Friends { get; set; }
+
+        public List<NewFriend> NewFriends { get; set; }
+
+        public List<UnTypedFriend> UnTypedFriends { get; set; }
+
+        public IODataInstanceAnnotationContainer InstanceAnnotations { get; set; }
+    }
+
+    [Flags]
+    public enum AccessLevel
+    {
+        Read = 1,
+        Write = 2,
+        Execute = 4
+    }
+
+    public enum Gender
+    {
+        Male = 1,
+        Female = 2
+    }
+
+    public enum Skill
+    {
+        CSharp,
+        Sql,
+        Web,
+    }
+
+    public enum Sport
+    {
+        Pingpong,
+        Basketball
+    }
+
+    public class Friend
+    {
+        [Key]
+        public int Id { get; set; }
+        [Required]
+        public string Name { get; set; }
+        [Range(10, 20)]
+        public int Age { get; set; }
+
+        public List<Order> Orders { get; set; }
+
+    }
+
+
+    public class Order
+    {
+        [Key]
+        public int Id { get; set; }
+
+        public int Price { get; set; }
+    }
+
+    public class NewFriend
+    {
+        [Key]
+        public int Id { get; set; }
+
+        public string Name { get; set; }
+
+        public int Age { get; set; }
+        public IODataInstanceAnnotationContainer InstanceAnnotations { get; set; }
+
+    }
+
+    public class UnTypedEmployee
```

Comment:
>UnTypedEmployee [](start = 17, length = 15)

what is the meaning "Untyped" here?

Reply:
Just gave a name to distinguish it to refer in untyped case

---
In reply to: [604629115](https://github.com/OData/WebApi/pull/2455#discussion_r604629115) [](ancestors = 604629115)